ert/data/record/_transformation.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+import tarfile
+import io
+import stat
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ert.data import RecordTransmitter
+
+_BIN_FOLDER = "bin"
+
+
+class RecordTransformation(ABC):
+    @abstractmethod
+    async def dump(
+        self, transmitter: "RecordTransmitter", mime: str, runpath: Path, location: Path
+    ) -> None:
+        raise NotImplementedError("not implemented")
+
+
+class FileRecordTransformation(RecordTransformation):
+    async def dump(
+        self, transmitter: "RecordTransmitter", mime: str, runpath: Path, location: Path
+    ) -> None:
+        _location = runpath / location
+        await transmitter.dump(_location, mime)
+
+
+class TarRecordTransformation(RecordTransformation):
+    async def dump(
+        self, transmitter: "RecordTransmitter", mime: str, runpath: Path, location: Path
+    ) -> None:
+        record = await transmitter._load_blob_record()
+        with tarfile.open(fileobj=io.BytesIO(record.data), mode="r") as tar:
+            tar.extractall(runpath / location)
+
+
+class ExecutableRecordTransformation(RecordTransformation):
+    async def dump(
+        self, transmitter: "RecordTransmitter", mime: str, runpath: Path, location: Path
+    ) -> None:
+        # pre-make bin folder if neccessary
+        Path(runpath / _BIN_FOLDER).mkdir(parents=True, exist_ok=True)
+
+        # create file(s)
+        await transmitter.dump(runpath / _BIN_FOLDER / location, mime)
+
+        # post-proccess if neccessary
+        path = runpath / _BIN_FOLDER / location
+        st = path.stat()
+        path.chmod(st.st_mode | stat.S_IEXEC)
+
+
+def path_to_bytes(file_path: Path) -> bytes:
+    tar_obj = io.BytesIO()
+    with tarfile.open(fileobj=tar_obj, mode="w") as tar:
```

Comment:
```suggestion
    with tarfile.open(fileobj=tar_obj, mode="wb") as tar:
```

Reply:
This way there's no encoding worries

Reply:
It needs to be either "r" or "w"

Reply:
No, it should be `wb`, which means you want to write binary data.

Reply:
writing tarfile is binary by default https://docs.python.org/3/library/tarfile.html